### PR TITLE
Fix for UserAgent.h issues

### PR DESCRIPTION
**Pull request by jmagman**
Identical to https://github.com/FirebaseExtended/flutterfire/pull/2099 but with the firebase_crashlytics CHANGELOG merge issue and analyzer warnings fixed.

Remove the UserAgent.h header in favor of defining `LIBRARY_VERSION` and `LIBRARY_NAME` in the GCC_PREPROCESSOR_DEFINITIONS in the podspec.

Fixes https://github.com/FirebaseExtended/flutterfire/issues/2087
Fixes https://github.com/FirebaseExtended/flutterfire/issues/1858
Fixes https://github.com/FirebaseExtended/flutterfire/issues/2084

### CODE REVIEW

**Comment by jmagman** [comment]
New analyzer warning for change introduced in #1686.  Fixing in this PR.  I guess the analyzer options aren't pegged to a version in this repo?
\cc @collinjackson

File: packages/cloud_firestore/cloud_firestore_platform_interface/test/method_channel_cloud_firestore_test.dart [hunk 1]
```diff
@@ -1028,7 +1028,6 @@ void main() {
                       'data': kMockDocumentSnapshotData,
                     },
                   },
-                  'source': 'default',
```
Comment (jmagman) [inline packages/cloud_firestore/cloud_firestore_platform_interface/test/method_channel_cloud_firestore_test.dart]: Analyzer warning fix: multiple 'source' keys.

File: packages/cloud_firestore/cloud_firestore_platform_interface/test/method_channel_cloud_firestore_test.dart [hunk 2]
```diff
@@ -1052,7 +1051,6 @@ void main() {
...
```
Comment (jmagman) [inline packages/cloud_firestore/cloud_firestore_platform_interface/test/method_channel_cloud_firestore_test.dart]: Analyzer warning fix: multiple 'source' keys.

**Comment by jmagman** [comment]
Manually merged with https://github.com/FirebaseExtended/flutterfire/commit/4d7b2bdd7fbdee65551754968b140653ea16128f to maintain authorship.  Closing.

**Comment by Saifallak** [comment]
## 0.15.5+3 not published yet, can you publish it ? @collinjackson

**Comment by awhitford** [comment]
Besides `firebase_auth`, I seem to also still have an issue with `firebase_storage`:
```
    /Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_storage-3.1.3/ios/Classes/FLTFirebaseStoragePlugin.m:6:9: fatal error: 'UserAgent.h' file not found
    #import "UserAgent.h"
            ^~~~~~~~~~~~~
    1 error generated.
```

**Comment by mono0926** [comment]
@jmagman 

Thanks, the build error was fixed 👍 
But, after this change, `bitcode_strip` started failed on iOS project which [bitcode enabled](https://github.com/flutter/flutter/wiki/Creating-an-iOS-Bitcode-enabled-app).


> 2020-03-16 02:13:56.970 xcodebuild[26510:54603] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/6q/wgy6jtp12w5gzgm9lzcglpqw0000gn/T/Runner_2020-03-16_02-13-56.965.xcdistributionlogs'.
error: exportArchive: bitcode_strip failed
> 
> Error Domain=IDEFoundationErrorDomain Code=1 "bitcode_strip failed" UserInfo={NSLocalizedDescription=bitcode_strip failed}

**Comment by jmagman** [comment]
@mono0926 please file a new issue at [flutter/flutter](https://github.com/flutter/flutter/issues/new/choose) and fill out the template so we can take a look.

**Comment by ifanchu** [comment]
so which version should I use? `0.15.5+2` still seeing same issue.

**Comment by DenisBogatirov** [comment]
Same UserAgent.h issue on version `0.15.5+2`, when `0.15.5+3` will be released?

**Comment by Saifallak** [comment]
finally it's released now `firebase-auth: 0.15.5+3`

**Pull request closed without merging**